docs/contributor/README.md

```diff
@@ -1 +1,62 @@
-In this folder, you can add any developer-related documentation, for example, advanced installation options, testing strategy, governance, etc.
\ No newline at end of file
+# Contributing to Kyma Metrics Collector
+
+## Overview
+
+To bill hyperscaler resources used by SKR clusters, the Kyma Control Plane (KCP) uses the Kyma Metrics Collector (KMC), which is integrated with Unified Metering using Event Data Platform (EDP).
+
+## Architecture
+
+Every SKR cluster runs in a hyperscaler account dedicated for the related global account, so it is shared between many clusters of the same customer. The hyperscaler account is payed by Kyma, and individual resource usage is charged to the customer. The bill to the end user contains one entry, listing the consumed capacity units (CU) without any further breakdown. The bill is created by the Unified Metering service.
+
+[!arch](./assets/arch.drawio.svg)
+
+The following step happens ones per every SKR registration and deregistration
+
+1. KEB registers/unregisters a new tenant in EDP
+
+The following steps happen periodically:
+
+2. KMC workers fetch the list of billable SKR clusters from [Kyma Environment Broker (KEB)](https://github.com/kyma-project/kyma-environment-broker/tree/main) and add them to a queue to work through them. If an error occurs, KMC re-queues the affected SKR cluster. For every process step, internal metrics are exposed with the [Prometheus client library](https://github.com/prometheus/client_golang). For details about the exposed metrics, see the [metrics.md](./metrics.md) file.
+2. KMC fetches the kubeconfig for every SKR cluster from the control plane resources.
+2. KMC fetches specific Kubernetes resources from the APIServer of every SKR cluster using the related kubeconfig.
+2. KMC maps the retrieved Kubernetes resources to a memory/CPU/storage value and sends the value to EDP as event stream.
+2. EDP calculates the consumed CUs based on the consumed CPU or storage with a fixed formula and sends the consumed CUs to Unified Metering.
+
+KMC fetches the amount of the following resource types from the SKR APIServer:
```

Comment:
why is this separate and not one of the numbered steps?

Reply:
done